library/src/context.cxx

```diff
@@ -0,0 +1,143 @@
+
+  vtksys::DynamicLoader::LibraryHandle handle = vtksys::DynamicLoader::OpenLibrary(libName);
+
+  if (!handle)
+  {
+    throw loading_exception("Cannot find GLX library");
+  }
+
+  using symbol = fptr (*)(const char*);
+
+  symbol address =
+    reinterpret_cast<symbol>(vtksys::DynamicLoader::GetSymbolAddress(handle, "glXGetProcAddress"));
+
+  if (!address)
+  {
+    throw symbol_exception("Cannot find glXGetProcAddress symbol");
+  }
+
+  return address;
+#else
+  throw loading_exception("Cannot use a GLX context on this platform");
+#endif
+}
+
+//----------------------------------------------------------------------------
+context::function context::egl()
+{
+#ifndef __APPLE__
+  std::string libName = vtksys::DynamicLoader::LibPrefix();
+  libName += "EGL";
+  libName += vtksys::DynamicLoader::LibExtension();
+
+  vtksys::DynamicLoader::LibraryHandle handle = vtksys::DynamicLoader::OpenLibrary(libName);
+
+  if (!handle)
+  {
+    throw loading_exception("Cannot find EGL library");
+  }
+
+  using symbol = fptr (*)(const char*);
+
+  symbol address =
+    reinterpret_cast<symbol>(vtksys::DynamicLoader::GetSymbolAddress(handle, "eglGetProcAddress"));
+
+  if (!address)
+  {
+    throw symbol_exception("Cannot find eglGetProcAddress symbol");
+  }
+
+  return address;
+#else
+  throw loading_exception("Cannot use a EGL context on this platform");
+#endif
+}
+
+//----------------------------------------------------------------------------
+context::function context::osmesa()
+{
+#ifndef __APPLE__
+  std::string libName = vtksys::DynamicLoader::LibPrefix();
+  libName += "OSMesa";
+  libName += vtksys::DynamicLoader::LibExtension();
+
+  vtksys::DynamicLoader::LibraryHandle handle = vtksys::DynamicLoader::OpenLibrary(libName);
+
+  if (!handle)
+  {
+    throw loading_exception("Cannot find OSMesa library");
+  }
+
+  using symbol = fptr (*)(const char*);
+
+  symbol address = reinterpret_cast<symbol>(
+    vtksys::DynamicLoader::GetSymbolAddress(handle, "OSMesaGetProcAddress"));
+
+  if (!address)
+  {
+    throw symbol_exception("Cannot find OSMesaGetProcAddress symbol");
+  }
+
+  return address;
+#else
+  throw loading_exception("Cannot use a OSMesa context on this platform");
+#endif
+}
+
+//----------------------------------------------------------------------------
+context::function context::wgl()
+{
+#ifdef _WIN32
+  std::string libName = vtksys::DynamicLoader::LibPrefix();
+  libName += "opengl32";
+  libName += vtksys::DynamicLoader::LibExtension();
+
+  vtksys::DynamicLoader::LibraryHandle handle = vtksys::DynamicLoader::OpenLibrary(libName);
+
+  if (!handle)
+  {
+    throw loading_exception("Cannot find opengl32 library");
+  }
+
+  using symbol = fptr (*)(const char*);
+
+  symbol address =
+    reinterpret_cast<symbol>(vtksys::DynamicLoader::GetSymbolAddress(handle, "wglGetProcAddress"));
+
+  if (!address)
+  {
+    throw symbol_exception("Cannot find wglGetProcAddress symbol");
+  }
```

Comment:
looks to me like you could easily factorize this in a single method with a `libName` and `functionName` param.